Scripts/Utilities/Credentials.cs

```diff
@@ -146,7 +147,7 @@ public Credentials(TokenOptions iamTokenOptions, string serviceUrl = null)
         #region SetCredentials
         private void SetCredentials(string username, string password, string url = null)
         {
-            if (username == APIKEY_AS_USERNAME)
+            if (username == APIKEY_AS_USERNAME && !password.StartsWith(ICP_PREFIX))
```

Comment:
can we add a test when using `ICP_Prefix`?

Reply:
Unfortunately Unity doesn't have a good testing framework. Right now I am only able to do integration tests with the services. I may be able to start testing now that modern .NET is available for Unity but I'll have to find some time to investigate this.